agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -1632,6 +1646,42 @@ bool monitor_thread::hal_event_handler(bwl::base_wlan_hal::hal_event_ptr_t event
 
         message_com::send_cmdu(slave_socket, cmdu_tx);
     } break;
+    case Event::RRM_Associated_STA_Link_Metrics_Response: {
+        auto metrics_result = static_cast<bwl::sAssociatedStaLinkMetricsResults *>(data);
+        auto response = message_com::create_vs_message<
+            beerocks_message::cACTION_MONITOR_CLIENT_ASSOCIATED_STA_LINK_METRIC_RESPONSE>(cmdu_tx);
+
+        if (!response) {
+            LOG(ERROR) <<
+                "Failed building cACTION_MONITOR_CLIENT_ASSOCIATED_STA_LINK_METRIC_RESPONSE msg";
+            return false;
+        }
+
+        memcpy(response->sta_mac().oct, metrics_result->sta_mac.oct,
```

Comment:
Don't use memcpy. IIRC we have an operator= for sMacAddr so you can just assign it; otherwise, use std::copy.

Reply:
BWL has a separate struct for the MAC address. `=` is not working because types are technically different.